LSP diagnostics was not working any more when launch zed from command line on windows 10

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When i start zed by command like `zed .` in command line on windows 10,so the LSP diagnostics of any language will not working , it mean that never show any error or warning messages whatever i typed.i checked log file,i think this line may be helpful: `2024-08-06T02:57:35.3325135+08:00 [ERROR] no worktree found for diagnostics path "xxx"`。

### Environment

Zed: v0.148.0 (Zed Dev 49e736d8ef7db2701df0a15c996a3a395bde23d8)
OS: Windows 10.0.19045
Memory: 7.9 GiB
Architecture: x86_64
GPU: Intel(R) HD Graphics 620 || Intel Corporation || Intel driver

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-08-06T03:17:00.3340125+08:00 [INFO] Creating buffer 0x25712605db8 of size 168224, name 'chunk-10', handle 12
2024-08-06T03:17:01.7211027+08:00 [INFO] Creating texture 0x25710f92a78 of size 1414x1024x1 and format R16Float, name 'atlas', handle 13
2024-08-06T03:17:01.842255+08:00 [INFO] Creating buffer 0x25712609558 of size 168224, name 'chunk-11', handle 14
2024-08-06T03:17:06.0034185+08:00 [INFO] Creating texture 0x25710f92068 of size 1452x1024x1 and format R16Float, name 'atlas', handle 15
2024-08-06T03:17:08.131722+08:00 [INFO] Creating buffer 0x257120b95c8 of size 229824, name 'chunk-12', handle 16
2024-08-06T03:17:09.0443793+08:00 [INFO] Creating buffer 0x257120ba678 of size 233520, name 'chunk-13', handle 17
2024-08-06T03:17:09.2440274+08:00 [INFO] Creating buffer 0x257120bcd68 of size 252560, name 'chunk-14', handle 18
2024-08-06T03:17:26.3815017+08:00 [INFO] Destroying texture 0x25710f8d9f8, handle 0
2024-08-06T03:17:26.3816186+08:00 [INFO] Destroying texture 0x25710f92a78, handle 13
2024-08-06T03:17:26.3818656+08:00 [INFO] Destroying texture 0x25710f92068, handle 15
2024-08-06T03:17:26.3821059+08:00 [INFO] Destroying buffer 0x257120be3a8, handle 1
2024-08-06T03:17:26.3823246+08:00 [INFO] Destroying buffer 0x2571260a608, handle 5
2024-08-06T03:17:26.3824228+08:00 [INFO] Destroying buffer 0x25712605298, handle 6
2024-08-06T03:17:26.3825119+08:00 [INFO] Destroying buffer 0x257120bcd68, handle 18
2024-08-06T03:17:26.3826004+08:00 [INFO] Destroying buffer 0x257120bff78, handle 2
2024-08-06T03:17:26.3826889+08:00 [INFO] Destroying buffer 0x25712608a38, handle 11
2024-08-06T03:17:26.3827771+08:00 [INFO] Destroying buffer 0x25712605db8, handle 12
2024-08-06T03:17:26.3828691+08:00 [INFO] Destroying buffer 0x2571260b128, handle 4
2024-08-06T03:17:26.3829568+08:00 [INFO] Destroying buffer 0x2571260ab98, handle 7
2024-08-06T03:17:26.3830448+08:00 [INFO] Destroying buffer 0x257120ba678, handle 17
2024-08-06T03:17:26.3831332+08:00 [INFO] Destroying buffer 0x25712607988, handle 3
2024-08-06T03:17:26.3832205+08:00 [INFO] Destroying buffer 0x25712609ae8, handle 8
2024-08-06T03:17:26.3833081+08:00 [INFO] Destroying buffer 0x2571260b6b8, handle 9
2024-08-06T03:17:26.3834555+08:00 [INFO] Destroying buffer 0x257120b95c8, handle 16
2024-08-06T03:17:26.3835439+08:00 [INFO] Destroying buffer 0x257126084a8, handle 10
2024-08-06T03:17:26.3836318+08:00 [INFO] Destroying buffer 0x25712609558, handle 14
2024-08-06T03:17:26.5500599+08:00 [ERROR] window not found
2024-08-06T03:17:26.5548947+08:00 [ERROR] oneshot canceled
2024-08-06T03:17:26.5573004+08:00 [ERROR] oneshot canceled
2024-08-06T03:17:26.5631002+08:00 [ERROR] oneshot canceled
2024-08-06T03:17:42.5170405+08:00 [INFO] ========== starting zed ==========
2024-08-06T03:17:42.6636026+08:00 [INFO] Use Microsoft YaHei UI as UI font.
2024-08-06T03:17:42.6656781+08:00 [INFO] Opening main db
2024-08-06T03:17:42.6692922+08:00 [INFO] Using git binary path: None
2024-08-06T03:17:42.6702854+08:00 [ERROR] SHELL environment variable is not assigned so we can't source login environment variables

Caused by:
    environment variable not found
2024-08-06T03:17:42.7745392+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-08-06T03:17:42.782105+08:00 [INFO] extensions updated. loading 1, reloading 0, unloading 0
2024-08-06T03:17:42.8045397+08:00 [ERROR] 找不到元素。 (0x80070490)
2024-08-06T03:17:42.819513+08:00 [INFO] Opening main db
2024-08-06T03:17:42.8227139+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-08-06T03:17:43.307264+08:00 [INFO] Enabling color space support
2024-08-06T03:17:43.4383306+08:00 [INFO] Adapter "Intel(R) HD Graphics 620"
2024-08-06T03:17:43.4384466+08:00 [INFO] No ray tracing extensions are supported
2024-08-06T03:17:43.4543915+08:00 [WARN] Requested size 1x1 is outside of surface capabilities
2024-08-06T03:17:43.4545635+08:00 [ERROR] No composite alpha flag for transparency: OPAQUE | INHERIT
2024-08-06T03:17:43.4547193+08:00 [INFO] Using surface present mode IMMEDIATE
2024-08-06T03:17:43.4996449+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-06T03:17:43.4999461+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-06T03:17:43.5001838+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-08-06T03:17:43.5044583+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.505487+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5074998+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5085715+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5166845+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5179244+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-08-06T03:17:43.5180868+08:00 [INFO] Skip function Some("to_device_position")
2024-08-06T03:17:43.5182267+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-08-06T03:17:43.5183655+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5187272+08:00 [INFO] Skip function Some("blend_color")
2024-08-06T03:17:43.5198282+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5204137+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5208008+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5215792+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5223886+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5229138+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5232986+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-08-06T03:17:43.5234096+08:00 [INFO] Skip function Some("to_device_position")
2024-08-06T03:17:43.5235166+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-08-06T03:17:43.5236237+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.5238187+08:00 [INFO] Skip function Some("blend_color")
2024-08-06T03:17:43.5504027+08:00 [ERROR] No composite alpha flag for transparency: OPAQUE | INHERIT
2024-08-06T03:17:43.550636+08:00 [INFO] Using surface present mode IMMEDIATE
2024-08-06T03:17:43.5751483+08:00 [ERROR] No composite alpha flag for transparency: OPAQUE | INHERIT
2024-08-06T03:17:43.5752551+08:00 [INFO] Using surface present mode IMMEDIATE
2024-08-06T03:17:43.624113+08:00 [INFO] Using surface present mode IMMEDIATE
2024-08-06T03:17:43.7448181+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-08-06T03:17:43.7500566+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7508246+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7517009+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7525224+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7534531+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.75413+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-08-06T03:17:43.7542827+08:00 [INFO] Skip function Some("to_device_position")
2024-08-06T03:17:43.7544256+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-08-06T03:17:43.7545688+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7548796+08:00 [INFO] Skip function Some("blend_color")
2024-08-06T03:17:43.7559342+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7566347+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7572731+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7584935+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.759744+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7607376+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7622484+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-08-06T03:17:43.7624501+08:00 [INFO] Skip function Some("to_device_position")
2024-08-06T03:17:43.7626701+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-08-06T03:17:43.7629468+08:00 [INFO] Skip function Some("to_tile_position")
2024-08-06T03:17:43.7635058+08:00 [INFO] Skip function Some("blend_color")
2024-08-06T03:17:43.7660918+08:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "Intel(R) HD Graphics 620", driver_name: "Intel Corporation", driver_info: "Intel driver" }
2024-08-06T03:17:43.7669525+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-08-06T03:17:43.7775109+08:00 [INFO] Opening main db
2024-08-06T03:17:43.7843509+08:00 [INFO] Using Windows API for pseudoconsole
2024-08-06T03:17:43.8303628+08:00 [INFO] Using Windows API for pseudoconsole
2024-08-06T03:17:43.9219955+08:00 [INFO] Using surface present mode IMMEDIATE
2024-08-06T03:17:44.0245103+08:00 [INFO] Creating texture 0x1daa674bf18 of size 1024x1024x1 and format R8Unorm, name 'atlas', handle 0
2024-08-06T03:17:44.0266606+08:00 [INFO] Creating buffer 0x1daa65b06b8 of size 65536, name 'chunk-0', handle 1
2024-08-06T03:17:44.1011834+08:00 [INFO] Creating buffer 0x1daa65b6548 of size 4096, name 'chunk-0', handle 2
2024-08-06T03:17:44.1014228+08:00 [INFO] Creating buffer 0x1daa65b38c8 of size 4144, name 'chunk-1', handle 3
2024-08-06T03:17:44.101602+08:00 [INFO] Creating buffer 0x1daa65b43e8 of size 9744, name 'chunk-2', handle 4
2024-08-06T03:17:44.1301323+08:00 [INFO] Creating buffer 0x1daa65b3338 of size 4096, name 'chunk-3', handle 5
2024-08-06T03:17:44.1308098+08:00 [INFO] Creating buffer 0x1daa65b4978 of size 4144, name 'chunk-4', handle 6
2024-08-06T03:17:44.1313526+08:00 [INFO] Creating buffer 0x1daa65b2da8 of size 9744, name 'chunk-5', handle 7
2024-08-06T03:17:45.0300258+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0304806+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.030634+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0310802+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0314706+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0318335+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0319781+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0320968+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0476312+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.0614463+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.1232317+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.123717+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.1238603+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.1239705+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.1240708+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.1241853+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.1242835+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.1243902+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.9894725+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:45.98979+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:46.1459673+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:46.3517395+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:46.554006+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:47.807045+08:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-08-06T03:17:47.807429+08:00 [ERROR] error fetching latest release

Caused by:
    0: request or operation took longer than the configured timeout time
    1: [28] Timeout was reached
2024-08-06T03:17:47.8095781+08:00 [INFO] Node runtime install_if_needed
2024-08-06T03:17:48.1774693+08:00 [INFO] starting language server. binary path: "C:\\Users\\K\\AppData\\Local\\Zed\\node\\node-v22.5.1-win-x64\\node.exe", working directory: "C:/", args: ["C:\\Users\\K\\AppData\\Local\\Zed\\copilot\\copilot-v0.5.0\\dist/agent.js", "--stdio"]
2024-08-06T03:17:49.1207125+08:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-05T19:17:49.102Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-08-05T19:17:49.102Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-08-06T03:17:49.1288341+08:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "834ff141-6c49-4576-b07a-6811816018a7",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-08-06T03:17:49.2481078+08:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-05T19:17:49.151Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-08-05T19:17:49.151Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-08-06T03:17:51.5401924+08:00 [INFO] Opening main db
2024-08-06T03:17:51.6396593+08:00 [INFO] Creating buffer 0x1daa65b1768 of size 14896, name 'chunk-6', handle 8
2024-08-06T03:17:51.6397975+08:00 [INFO] Creating buffer 0x1daa65b5a28 of size 37968, name 'chunk-7', handle 9
2024-08-06T03:17:51.6462527+08:00 [INFO] Initializing default prettier with plugins {}
2024-08-06T03:17:51.6464263+08:00 [INFO] starting language servers for TypeScript: vtsls, eslint
2024-08-06T03:17:51.6465589+08:00 [INFO] starting language server "vtsls", path: "\\\\?\\D:\\Codes\\t\\fastify", id: 1
2024-08-06T03:17:51.6468602+08:00 [INFO] starting language server "eslint", path: "\\\\?\\D:\\Codes\\t\\fastify", id: 2
2024-08-06T03:17:51.6474416+08:00 [INFO] Initializing default prettier with plugins {}
2024-08-06T03:17:51.6475842+08:00 [INFO] starting language servers for TypeScript: vtsls, eslint
2024-08-06T03:17:51.6494364+08:00 [INFO] fetching latest version of language server "vtsls"
2024-08-06T03:17:51.6495461+08:00 [INFO] Node runtime install_if_needed
2024-08-06T03:17:51.690057+08:00 [INFO] fetching latest version of language server "eslint"
2024-08-06T03:17:51.69021+08:00 [INFO] downloading language server "eslint"
2024-08-06T03:17:51.7424935+08:00 [INFO] Creating buffer 0x1daa65b0128 of size 37968, name 'chunk-8', handle 10
2024-08-06T03:17:52.0603982+08:00 [INFO] Node runtime install_if_needed
2024-08-06T03:17:52.1549633+08:00 [INFO] Creating buffer 0x1daa65b0c48 of size 15008, name 'chunk-9', handle 11
2024-08-06T03:17:52.15526+08:00 [INFO] Creating buffer 0x1daa65b11d8 of size 37968, name 'chunk-10', handle 12
2024-08-06T03:17:52.4119184+08:00 [INFO] Creating buffer 0x1daa65b2818 of size 38752, name 'chunk-11', handle 13
2024-08-06T03:17:52.6309613+08:00 [INFO] starting language server. binary path: "C:\\Users\\K\\AppData\\Local\\Zed\\node\\node-v22.5.1-win-x64\\node.exe", working directory: "\\\\?\\D:\\Codes\\t\\fastify", args: ["C:\\Users\\K\\AppData\\Local\\Zed\\languages\\eslint\\vscode-eslint-release/2.4.4\\vscode-eslint/server/out/eslintServer.js", "--stdio"]
2024-08-06T03:17:53.0656471+08:00 [INFO] Language server with id 2 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "ESLint server running in node v22.5.1"
}
2024-08-06T03:17:53.084746+08:00 [WARN] unhandled capability registration: Registration { id: "f047b898-436f-4d61-bf89-66e5f54c5424", method: "workspace/didChangeConfiguration", register_options: Some(Object {}) }
2024-08-06T03:17:53.0910365+08:00 [WARN] unhandled capability registration: Registration { id: "dd267e98-bb56-4988-8f12-b0a68df904d4", method: "workspace/didChangeWorkspaceFolders", register_options: Some(Object {}) }
2024-08-06T03:17:53.1144108+08:00 [INFO] Language server with id 2 sent unhandled notification eslint/noLibrary:
{
  "source": {
    "uri": "file:///D:/Codes/t/fastify/main.ts"
  }
}
2024-08-06T03:17:53.9110213+08:00 [WARN] request completed with error: failed to connect to the server
2024-08-06T03:17:54.9831449+08:00 [INFO] Node runtime install_if_needed
2024-08-06T03:17:57.8483476+08:00 [INFO] downloading language server "vtsls"
2024-08-06T03:17:57.8569223+08:00 [INFO] Node runtime install_if_needed
2024-08-06T03:17:58.2255638+08:00 [INFO] starting language server. binary path: "C:\\Users\\K\\AppData\\Local\\Zed\\node\\node-v22.5.1-win-x64\\node.exe", working directory: "\\\\?\\D:\\Codes\\t\\fastify", args: ["C:\\Users\\K\\AppData\\Local\\Zed\\languages\\vtsls\\node_modules/@vtsls/language-server/bin/vtsls.js", "--stdio"]
2024-08-06T03:18:06.8009822+08:00 [ERROR] no worktree found for diagnostics path "d:\\Codes\\t\\fastify\\tsconfig.json"
2024-08-06T03:18:06.8011813+08:00 [ERROR] no worktree found for diagnostics path "d:\\Codes\\t\\fastify\\tsconfig.json"
2024-08-06T03:18:07.8465832+08:00 [ERROR] no worktree found for diagnostics path "d:\\Codes\\t\\fastify\\main.ts"
2024-08-06T03:18:08.9129173+08:00 [INFO] Creating buffer 0x1daa65b1cf8 of size 38080, name 'chunk-12', handle 14
2024-08-06T03:18:09.5620074+08:00 [ERROR] no worktree found for diagnostics path "d:\\Codes\\t\\fastify\\main.ts"
2024-08-06T03:18:11.2161519+08:00 [WARN] Generic lsp request to vtsls failed: Request textDocument/completion failed with message: Reduce of empty array with no initial value
2024-08-06T03:18:11.6814785+08:00 [ERROR] no worktree found for diagnostics path "d:\\Codes\\t\\fastify\\main.ts"
2024-08-06T03:18:11.9970793+08:00 [INFO] Waiting for default prettier to install
2024-08-06T03:18:11.9978303+08:00 [INFO] Starting prettier at path "C:\\Users\\K\\AppData\\Local\\Zed\\prettier"
2024-08-06T03:18:11.9987451+08:00 [INFO] Node runtime install_if_needed
2024-08-06T03:18:12.4490398+08:00 [INFO] starting language server. binary path: "C:\\Users\\K\\AppData\\Local\\Zed\\node\\node-v22.5.1-win-x64\\node.exe", working directory: "C:\\Users\\K\\AppData\\Local\\Zed\\prettier", args: ["C:\\Users\\K\\AppData\\Local\\Zed\\prettier\\prettier_server.js", "C:\\Users\\K\\AppData\\Local\\Zed\\prettier"]
2024-08-06T03:18:12.4979261+08:00 [ERROR] no worktree found for diagnostics path "d:\\Codes\\t\\fastify\\main.ts"
2024-08-06T03:18:12.8126381+08:00 [INFO] Started default prettier in "C:\\Users\\K\\AppData\\Local\\Zed\\prettier"

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/zed/src/main.rs
+++ b/crates/zed/src/main.rs
@@ -480,13 +480,11 @@ fn main() {
         let prompt_builder = init_common(app_state.clone(), cx);
 
         let args = Args::parse();
-        println!("==> Args: {:?}", args);
         let urls: Vec<_> = args
             .paths_or_urls
             .iter()
             .filter_map(|arg| parse_url_arg(arg, cx).log_err())
             .collect();
-        println!("==> Urls: {:?}", urls);
 
         if !urls.is_empty() {
             open_listener.open_urls(urls)

--- a/crates/zed/src/main.rs
+++ b/crates/zed/src/main.rs
@@ -480,11 +480,13 @@ fn main() {
         let prompt_builder = init_common(app_state.clone(), cx);
 
         let args = Args::parse();
+        println!("==> Args: {:?}", args);
         let urls: Vec<_> = args
             .paths_or_urls
             .iter()
             .filter_map(|arg| parse_url_arg(arg, cx).log_err())
             .collect();
+        println!("==> Urls: {:?}", urls);
 
         if !urls.is_empty() {
             open_listener.open_urls(urls)
@@ -1048,7 +1050,10 @@ struct Args {
 
 fn parse_url_arg(arg: &str, cx: &AppContext) -> Result<String> {
     match std::fs::canonicalize(Path::new(&arg)) {
-        Ok(path) => Ok(format!("file://{}", path.to_string_lossy())),
+        Ok(path) => Ok(format!(
+            "file://{}",
+            path.to_string_lossy().trim_start_matches(r#"\\?\"#)
+        )),
         Err(error) => {
             if arg.starts_with("file://")
                 || arg.starts_with("zed-cli://")

